lighthouse-core/audits/consistently-interactive.js

```diff
@@ -0,0 +1,248 @@
+        .filter(isLongEnoughQuietPeriod);
+
+    const cpuQueue = cpuQuietPeriods.slice();
+    const networkQueue = networkQuietPeriods.slice();
+
+    // We will check for a CPU quiet period contained within a Network quiet period or vice-versa
+    let cpuCandidate = cpuQueue.shift();
+    let networkCandidate = networkQueue.shift();
+    while (cpuCandidate && networkCandidate) {
+      if (cpuCandidate.start >= networkCandidate.start) {
+        // CPU starts later than network, it must be contained by network or we check the next network
+        if (networkCandidate.end >= cpuCandidate.start + REQUIRED_QUIET_WINDOW) {
+          return {
+            cpuQuietPeriod: cpuCandidate,
+            networkQuietPeriod: networkCandidate,
+            cpuQuietPeriods,
+            networkQuietPeriods,
+          };
+        } else {
+          networkCandidate = networkQueue.shift();
+        }
+      } else {
+        // Network starts later than CPU, it must be contained by CPU or we check the next CPU
+        if (cpuCandidate.end >= networkCandidate.start + REQUIRED_QUIET_WINDOW) {
+          return {
+            cpuQuietPeriod: cpuCandidate,
+            networkQuietPeriod: networkCandidate,
+            cpuQuietPeriods,
+            networkQuietPeriods,
+          };
+        } else {
+          cpuCandidate = cpuQueue.shift();
+        }
+      }
+    }
+
+    const culprit = cpuCandidate ? 'Network' : 'CPU';
+    throw new Error(`${culprit} did not quiet for at least 5s before the end of the trace.`);
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!Promise<!AuditResult>}
+   */
+  static audit(artifacts) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const computedArtifacts = [
+      artifacts.requestNetworkRecords(devtoolsLog),
+      artifacts.requestTracingModel(trace),
+      artifacts.requestTraceOfTab(trace),
+    ];
+
+    return Promise.all(computedArtifacts)
+      .then(([networkRecords, traceModel, traceOfTab]) => {
+        if (!traceOfTab.timestamps.firstMeaningfulPaint) {
+          throw new Error('No firstMeaningfulPaint found in trace.');
+        }
+
+        const longTasks = TracingProcessor.getMainThreadTopLevelEvents(traceModel, trace)
+            .filter(event => event.end - event.start >= 50);
+        const quietPeriodInfo = this.findOverlappingQuietPeriods(longTasks, networkRecords,
+            traceOfTab);
+        const cpuQuietPeriod = quietPeriodInfo.cpuQuietPeriod;
+
+        const timestamp = Math.max(
+          cpuQuietPeriod.start,
+          traceOfTab.timestamps.firstMeaningfulPaint,
+          traceOfTab.timestamps.domContentLoaded
+        ) * 1000;
+        const timeInMs = (timestamp - traceOfTab.timestamps.navigationStart * 1000) / 1000;
+        const extendedInfo = Object.assign(quietPeriodInfo, {timestamp, timeInMs});
```

Comment:
that is...not a terribly useful type definition :P If I wanted to use that `extendedInfo` I would have to check the `param` signatures of `_findCPUQuietPeriods` and `_findNetworkQuietPeriods` and read `findOverlappingQuietPeriods` to make sure there's only one code escape path. We don't have a great place for documenting artifact shapes, but it would be nice to always do it, even if in an awkward place